includes/data-events/connectors/class-esp-connector.php

```diff
@@ -140,7 +140,18 @@ public static function subscription_updated( $timestamp, $data, $client_id ) {
 			return;
 		}
 
-		self::sync( $contact, sprintf( 'RAS Woo Subscription updated. Status changed from %s to %s', $data['status_before'], $data['status_after'] ) );
+		// Schedule a backup sync in two minutes if the subscription is being reactivated, to get around potential race conditions.
+		$delay = 'active' === $data['status_after'] ? 120 : 0;
+
+		self::sync(
+			$contact,
+			sprintf(
+				'RAS Woo Subscription updated. Status changed from %s to %s',
```

Comment:
Let's add something to the message to identify these requests

Reply:
I was not able to test this yet. But a few comments/questions

Here it seems you are replacing the sync with a scheduled sync. So it's not a backup sync. I would leave the sync as is and schedule one additional backup sync.

Also, I'm not sure about triggering this only when a subscription is set to active. A renewal could end up with a subscription in another state, and the race condition can also happen in that case.

I think we need to identify situations when two syncs are triggered too close to one another, and then schedule a third one just to be sure. (or if there's an easy and safe way, delay the second one as you did here. But the second one is not necessarily settingg the status to active)

Reply:
> Here it seems you are replacing the sync with a scheduled sync. So it's not a backup sync. I would leave the sync as is and schedule one additional backup sync.

It doesn't just delay the sync, it actually schedules another sync after completing the regular sync (`upsert` [call here](https://github.com/Automattic/newspack-plugin/pull/3603/files#diff-54a027d8960b76963f30ce437f31ebe5e011ea94e6860c2b60423378f5330731R114)).

> Also, I'm not sure about triggering this only when a subscription is set to active. A renewal could end up with a subscription in another state, and the race condition can also happen in that case.

We can certainly add a backup sync in other instances, too—we would just need to add that third param to the `sync` call to trigger the schedule after the given number of seconds.

Reply:
> Let's add something to the message to identify these requests

I did [add a log](https://github.com/Automattic/newspack-plugin/pull/3603/files#diff-54a027d8960b76963f30ce437f31ebe5e011ea94e6860c2b60423378f5330731R118) whenever a backup sync is scheduled. Maybe these logs should be done through the `newspack_log` action instead?

Reply:
> We can certainly add a backup sync in other instances, too—we would just need to add that third param to the sync call to trigger the schedule after the given number of seconds.

wonder if we can identify when a renewal happens. And then schedule a few minutes after the renew is done

Reply:
3e28d39 adds an optional `$data` param to the Sync class's logger method. If it receives any data, it will log it to `newspack_log`.